ambry-router/src/main/java/com.github.ambry.router/GetBlobOperation.java

```diff
@@ -560,10 +592,12 @@ private void fetchRequests(RequestRegistrationCallback<GetOperation> requestRegi
      * Check if the operation on the chunk is eligible for completion, if so complete it.
      */
     void checkAndMaybeComplete() {
-      if (chunkOperationTracker.isDone()) {
+      if (!chunkCompleted && chunkOperationTracker.isDone()) {
         if (chunkOperationTracker.hasSucceeded()) {
-          // override any previously set exceptions.
-          chunkException = null;
+          // override any previously set exceptions
+          if (chunkException != null) {
```

Comment:
I don't remember why I changed that.  I'll get rid of it because it's pretty useless, though.